### DISCUSSION TITLE
Yup with useFieldArray and yup.test() validation error missing

### DISCUSSION THREAD

**Discussion by mcjanowski** [react-hook-form/react-hook-form#2641] [Q&A]
Hello,

I tried to display errors for local object inside array using yup schema validation and .test() method. Validation runs inside yup but RHF does not receive error object. 
In case of this example let's say we have array of objects (firstName, lastName) and only firstName or lastName (not both) may be set. Errors displays after press "submit" only. I would like to update errors after input change (according to mode in useForm). It would be best if I would not be obligated to use "trigger" method in every onChange input. 
What is weird that yup validation runs after every change but RHF errors are still empty. I noticed that it might be something related to path inside "this.createError" like RHF can not find field (but it's not field related error - at least not only one). I've tried to change path to something like 'test[0].firstName' and it works like it should but then if I delete lastName - error msg remains so it works only for firstName. 
Please advise. 

Codesandbox example bellow:
https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-uc9by

**Comment by bluebill1049** [comment]
i believe this one is already reported and WIP: https://github.com/react-hook-form/resolvers/issues/43

**Reply by mcjanowski** [reply]
Thanks,
...but seems problem remains the same. Update on input does not change errors dict - only submit button force to reload errors

**Reply by bluebill1049** [reply]
That's what I meant above, it's a bug.

Try out this beta: @hookform/resolvers@0.0.7-beta.2 let me know if this issue is resolved.

**Reply by mcjanowski** [reply]
Looks like there is no difference using 0.0.7-beta.2. Still errors applies after submit only.

**Reply by bluebill1049** [reply]
`firstOrLast` doesn't even exist in the form, you will have to use `trigger()` to show that `error`

**Reply by mcjanowski** [reply]
In order to display error I need to add "dummy" input  or put trigger on every onChange input ? May you show me working example ?

https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-8444v

There is no way to simply put into errors everything what yup returns ?

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-hk2u3

**Reply by mcjanowski** [reply]
Thank you 
but it still does not work. Put firstName then put LastName - no error message untll Submit.
Same happens with array.min(2) when you append second row: error "test field must have at least 2 items" remains

**Reply by bluebill1049** [reply]
Try this example: https://codesandbox.io/s/react-hook-form-fieldsarray-yup-validation-min-length-57qtb?file=/src/index.js

**Reply by mcjanowski** [reply]
Same - does not work in both cases (for min(2) and firstOrLast). Errors appears only after submit

https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-9ibgj

What I simply need is to display errors for Object level validation (one person) and Array level validation(many persons, like there is more than 1). Single field level validation works very well.

**Reply by bluebill1049** [reply]
react hook form is focused on field-level validation, with your use case, you will need to trigger the validation at the useEffect the example demonstrates that or build your custom validation with resovler.

**Reply by mcjanowski** [reply]
I've applied useEffect example but it does not work. 
https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-9ibgj

Would you be able to modify above codesandbox with working "firstOrLast" error ?

**Reply by mcjanowski** [reply]
I've created own resolver on top of yupResolver, changed only line where it returns mapped errors to fields - I've forced to return all
![image](https://user-images.githubusercontent.com/2385412/90784672-cfdc1180-e301-11ea-9352-ee70b882ab8c.png)
into
![image](https://user-images.githubusercontent.com/2385412/90784710-d9fe1000-e301-11ea-863c-29dce8bf6b61.png)

so resolver returns errors correctly but still errors dict in userForm is empty... 
![image](https://user-images.githubusercontent.com/2385412/90785064-3a8d4d00-e302-11ea-88a8-818a85596fc8.png)

How may I force useForm 'errors' to display all what resolver returns ?

**Reply by bluebill1049** [reply]
`trigger` will force the entire form errors

**Reply by bluebill1049** [reply]
can you share your CSB?

**Reply by mcjanowski** [reply]
Sure,

https://codesandbox.io/s/react-hook-form-usefieldarray-template-forked-zm0wp
